### PR TITLE
Refactor basic String classes

### PR DESCRIPTION
**Pull request by ksh8281**
* Implement String with inline buffer. this reduce malloc count & reduce memory usage
* Remove typed-gc-malloc from some classes.
  typed malloc classes should have storage for descriptor each malloc.

Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

**Comment by ksh8281** [comment]
```
function test()
{
    for (var t = 0; t < 1000000; t ++) {
        var str = "";
        for (var i = 0; i < 17; i ++) {
            str += i;
        }
    }
    print(str.length)
}

var before = new Date()
test()
var after = new Date()
print(after - before)
```
1551ms -> 1444ms

**Comment by ksh8281** [comment]
octane 1667->1716

File: src/runtime/String.cpp [hunk 1]
```diff
@@ -645,6 +717,22 @@ String* String::fromUTF8ToCompressibleString(VMInstance* instance, const char* s
 }
 #endif
 
+String* String::fromCharCode(char32_t code)
+{
+    if (code < 128) {
+        char c = (char)code;
+        return new ASCIIStringWithInlineBuffer(&c, 1);
+    } else if (code <= 0x10000) {
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/String.cpp]: Good point! I've fixed that part

[the remark this answers is not in the document]
Comment (seungsoo47) [reply src/runtime/String.cpp]: Huh~ I'm glad it's OK. ^^

**Review by bbrto21 — APPROVED** [review]

**Review by seungsoo47 — APPROVED** [review]
👍

**Pull request merged**